tt_eager/tensor/tensor.hpp

```diff
@@ -35,6 +35,7 @@ struct Tensor {
         std::vector<bool> tensor_populated = {};
         uint32_t main_thread_ref_count = 0;
         std::atomic<uint32_t> num_sibling_workers_sharing_tensor = 0;
+        std::atomic<bool> main_thread_tensor = true;
```

Comment:
@tt-asaigal will you be able to ramp me up on synchronizations situation in Tensor?